### PR TITLE
Cleanup ISmsService

### PR DESCRIPTION
**Pull request by MikeAlhayek**

### CODE REVIEW

**Comment by hishamco** [comment]
While you working on SMS again, I'm still worrying why you didn't use `KeyedServices` instead of resolving different providers yourself?!!

**Comment by MikeAlhayek** [comment]
Because with keyed-services, you can't do this 

https://github.com/OrchardCMS/OrchardCore/blob/e759e2cb984e2e25a3542a88dfef192d62876614/src/OrchardCore.Modules/OrchardCore.Sms/Drivers/SmsSettingsDisplayDriver.cs#L52-L53 

You don't have access to the keys.

**Comment by hishamco** [comment]
That's why we need the PR #15100. IMHO let us:

1. Revert the PR changes
2. Use the strategy pattern as you did before, but using the keyed services under the hood

**Comment by MikeAlhayek** [comment]
I won't want to use keyed service for this.

This has an advantage over keyed service where we don't need to add extra services to the DI container since these provider are resolved without the being registered into the container. 

Also, I don't like that we have to implement a workaround for keyed-services in OC.

In this PR, I am just doing some cleanup to simplify the service

**Comment by hishamco** [comment]
> This approach has an advantage over keyed service where we don't need to add extra services to the DI container since these provider are resolved without the being registered into the container. Also, it has the IsEnable flag for each service where one can enable/disable provider.

There's no extra service, what you are done is keeping the provider name in the options itself, which is the first time I saw in the Orchard Core code-base since the beginning

> Also, I don't like that we have to implement a workaround for keyed-services in OC.

This is another thing, we already did something similar in Localization while not supporting ASP.NET Core, don't forget we can't run ASP.NET Core team in something we need

File: src/OrchardCore/OrchardCore.Sms.Core/Services/SmsService.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@ namespace OrchardCore.Sms.Services;
 public class SmsService : ISmsService
 {
     private readonly ISmsProviderResolver _smsProviderResolver;
+    private ISmsProvider _provider;
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Sms.Core/Services/SmsService.cs:9]:
```suggestion
    private ISmsProvider _provider;

    private readonly ISmsProviderResolver _smsProviderResolver;
```

**Review by hishamco — APPROVED** [review]

File: src/OrchardCore/OrchardCore.Sms.Core/Services/SmsService.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Sms.Core/Services/SmsService.cs:9]: You're right

**Pull request merged**